calib/hmap.h

```diff
@@ -0,0 +1,58 @@
+/*
+ * Phoenix-Pilot
+ *
+ * hmap.c
+ * 
+ * String keyed hashmap for pointer storage implementation.
+ *
+ * Copyright 2022 Phoenix Systems
+ * Author: Mateusz Niewiadomski
+ *
+ * This file is part of Phoenix-Pilot software
+ *
+ * %LICENSE%
+ */
+
+#include <stdlib.h>
+
+/* Hashmap entry representing key/value pair */
+typedef struct _hmap_entry_t {
+	const char *key;    /* key of a hashmap entry */
+	void *value;        /* value referenced by 'key' */
+	unsigned long hash; /* (hashed key) mod (hasmap capacity) */
+} hmap_entry_t;
+
+
+/* Hashmap metadata structure */
+typedef struct {
+	hmap_entry_t *list; /* array of hashmap elements */
+	size_t nitems;      /* maximum capacity of hashmap */
+	size_t used;        /* used capacity of hashmap */
```

Comment:
The common way to name these fields for containers (e.x. C++ stl):
```suggestion
	size_t capacity;      /* maximum capacity of hashmap */
	size_t size;        /* used capacity of hashmap */
```